internal/controller/serverset/serverset.go

```diff
@@ -176,6 +181,48 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	}, nil
 }
 
+func didNrIncrease(cr *v1alpha1.ServerSet, replicas []v1alpha1.Server) bool {
+	return len(replicas) > cr.Status.AtProvider.Replicas
+}
+
+func didNrDecrease(cr *v1alpha1.ServerSet, replicas []v1alpha1.Server) bool {
+	return len(replicas) < cr.Status.AtProvider.Replicas
+}
+
+func didNrOfReplicasChange(cr *v1alpha1.ServerSet, replicas []v1alpha1.Server) bool {
+	return didNrIncrease(cr, replicas) || didNrDecrease(cr, replicas)
+}
+
+func (e *external) populateCRStatus(cr *v1alpha1.ServerSet, serverSetReplicas []v1alpha1.Server) {
+	if cr.Status.AtProvider.ReplicaStatuses == nil || didNrOfReplicasChange(cr, serverSetReplicas) {
+		cr.Status.AtProvider.ReplicaStatuses = make([]v1alpha1.ServerSetReplicaStatus, cr.Spec.ForProvider.Replicas)
+	}
+	cr.Status.AtProvider.Replicas = len(serverSetReplicas)
+
+	for i, replica := range serverSetReplicas {
+		replicaStatus := computeStatus(serverSetReplicas[i].Status.AtProvider.State)
+		cr.Status.AtProvider.ReplicaStatuses[i] = v1alpha1.ServerSetReplicaStatus{
+			Name:         replica.Name,
```

Comment:
You can just use serverSetReplicas[i].Name instead and not have replica in the for loop at all